test/integration/markers.cc

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2023 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <gtest/gtest.h>
+
+#include <gz/common/Console.hh>
+#include <gz/common/Util.hh>
+#include <gz/rendering/RenderEngine.hh>
+#include <gz/rendering/RenderingIface.hh>
+#include "gz/rendering/Scene.hh"
+#include <gz/transport/Node.hh>
+#include <gz/utilities/ExtraTestMacros.hh>
+
+#include "gz/sim/rendering/MarkerManager.hh"
+#include "gz/sim/Server.hh"
+#include "gz/sim/test_config.hh"
+
+#include "plugins/MockSystem.hh"
+#include "../helpers/EnvTestFixture.hh"
+
+using namespace gz;
+using namespace gz::sim;
+
+/// \brief Test MarkersTest system
+class MarkersTest : public InternalFixture<::testing::Test>
+{
+};
+
+std::mutex mutex;
+std::vector<msgs::Marker> markerMsgs;
+
+/////////////////////////////////////////////////
+void markerCb(const msgs::Marker &_msg)
+{
+  mutex.lock();
+  markerMsgs.push_back(_msg);
+  mutex.unlock();
+}
+
+/////////////////////////////////////////////////
+TEST_F(MarkersTest, MarkerPublisher)
+{
+  std::map<std::string, std::string> params;
+  auto engine = ignition::rendering::engine("ogre2", params);
+  auto scene = engine->CreateScene("testscene");
```

Comment:
suggestion to use `rendering::sceneFromFirstRenderEngine` so the test is not specific to `ogre2`, [example here](https://github.com/gazebosim/gz-sim/blob/e2a5c7bc937a5ea2de7c53e10c293949913e258f/src/systems/camera_video_recorder/CameraVideoRecorder.cc#L281)

Reply:
That didn't seem to work. It returns a nullptr.

Reply:
@iche033 , any suggestion here or should I keep the existing code?

Reply:
ah it looks like `sceneFromFirstRenderEngine` will rely on the test spinning up the sensors system first. We can just keep things simple as it is now.